src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -294,6 +294,17 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
             return $a == $b ? 0 : 1;
         });
 
+        // force string for an object. See https://bugs.php.net/bug.php?id=66966
+        foreach ($extra as $paramName => $param) {
+            if (!\is_object($param)) {
+                continue;
+            }
+            if (!\is_callable([$param, '__toString'])) {
```

Comment:
I think a future PR can deal with if you pass an array or callable. 
Or we may think that is not needed.